catroid/src/main/java/org/catrobat/catroid/formulaeditor/InternToExternGenerator.java

```diff
@@ -319,6 +308,55 @@ private String generateExternStringFromToken(InternToken internToken, boolean tr
 		}
 	}
 
+	private String generateExternNumberStringForToken(String number, boolean trimNumbers) {
+
+		if (trimNumbers) {
+			number = getDesiredNumberRepresentation(number);
+		}
+
+		if (!number.contains(".")) {
+			return number;
+		}
+
+		String left = number.substring(0, number.indexOf('.'));
+		String right = number.substring(number.indexOf('.') + 1);
+
+		return left + getExternStringForInternTokenValue(".", context) + right;
+	}
+
+	private String getDesiredNumberRepresentation(String number) {
+
+		Double exponentRepresentation = Double.parseDouble(number);
+		number = String.valueOf(exponentRepresentation);
+
+		if (!checkContainsExponentAndSmallerOne(number)) {
```

Comment:
you could just go with 
if(number.contains("E") && number.contains("-")) { .... }

or probably even 
if(number.contains("E-")) {

but i guess the clean code version would be 
if(exponentRepresentation < 1 && number.contains("E"))
to make clear to the reader for what condition youre actually checking.

also exponentRepresentation is kind of an odd name, because "value" would be sufficient and not misleading (the double exponentRepresentation doesnt have anything todo with exponentRepresentation, its just a double...)